pkg/genericapiserver/config_selfclient.go

```diff
@@ -87,12 +89,18 @@ func (s *SecureServingInfo) NewSelfClientConfig(token string) (*restclient.Confi
 		}
 
 		if (net.ParseIP(host) != nil && certMatchesIP(x509Cert, host)) || certMatchesName(x509Cert, host) {
-			derCA = s.Cert.Certificate[0]
+			derCA, err = findCAinChain(s.Cert)
```

Comment:
in the case where a cert is being used from a publicly issued CA, the root may be omitted from the chain, depending on clients to have the root in their trusted bundles. Rather than erroring in that case, we should probably log a warning and fall back to using the system roots.